solutions/cses-1161.mdx

```diff
@@ -20,13 +20,14 @@ If you're wondering why Huffman Coding always produces an optimal solution, see
 
 ## Implementation
 
+As we want to selete both shortest sticks and then insert the combined new stick, we will want to have a data structure where we can get the smallest value in the collection and add new value to it. The insert operation can be done easily with a priority queue in $\mathcal{O}(\log n)$ time while getting the smallest value takes constant time.
```

Comment:
I thought getting the smallest value from a priority queue also takes O(logN) time?

Reply:
Naya, *getting* the smallest element takes constant time, but deleting it will take logarithmic time as the heap is modified. 
`Implementation note: this implementation provides O(log(n)) time for the enqueuing and dequeuing methods (offer, poll, remove() and add); linear time for the remove(Object) and contains(Object) methods; and constant time for the retrieval methods (peek, element, and size).`
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/PriorityQueue.html

Reply:
My bad. Thanks for the information.

Reply:
when we get it, don't we also have to pop it from the queue or smth

Reply:
remove() or poll() in Java does two things: get the top element, and remove it. Getting it takes O(0) and removing it takes O(logn).  If we only look at the method, then ya, these methods take logarithmic time.